src/Compilers/CSharp/Portable/Symbols/Synthesized/GeneratedNames.cs

```diff
@@ -54,17 +54,34 @@ internal static string MakeLambdaDisplayClassName(int methodOrdinal, int generat
             return MakeMethodScopedSynthesizedName(GeneratedNameKind.LambdaDisplayClass, methodOrdinal, generation, suffix: "DisplayClass", entityOrdinal: closureOrdinal, entityGeneration: closureGeneration, isTypeName: true);
         }
 
-        internal static string MakeAnonymousTypeTemplateName(int index, int submissionSlotIndex, string moduleId)
+        private static string MakeTopLevelTypeName(string typePostfix, int index, int submissionSlotIndex, string moduleId)
         {
-            var name = "<" + moduleId + ">f__AnonymousType" + StringExtensions.GetNumeral(index);
-            if (submissionSlotIndex >= 0)
+            var name = "<" + moduleId + typePostfix + StringExtensions.GetNumeral(index);
```

Comment:
Will `typePostfix` always begin with ">"?  Should that be handled here, rather than in each caller?